### ISSUE TITLE
TypeError: __init__() got an unexpected keyword argument 'transform'

### ISSUE THREAD

**Issue by IncubatorShokuhou** [PyTorchLightning/lightning-flash#1255]
## 🐛 Bug
```
Uncaught exception
Traceback (most recent call last):
  File "/mnt/data/buduanban_detection/meteo.py", line 103, in <module>
    datamodule = ObjectDetectionData.from_coco(
  File "/data/miniconda3/envs/torch/lib/python3.9/site-packages/flash/image/detection/data.py", line 386, in from_coco
    return cls.from_icedata(
  File "/data/miniconda3/envs/torch/lib/python3.9/site-packages/flash/image/detection/data.py", line 220, in from_icedata
    return cls(
TypeError: __init__() got an unexpected keyword argument 'transform'
```

### To Reproduce

The code is directly copied from https://lightning-flash.readthedocs.io/en/latest/reference/object_detection.html

#### Code sample
```python
from dataclasses import dataclass
import albumentations as alb
from icevision.tfms import A

from flash import InputTransform
from flash.core.integrations.icevision.transforms import IceVisionTransformAdapter
from flash.image import ObjectDetectionData


@dataclass
class BrightnessContrastTransform(InputTransform):
    image_size: int = 128

    def per_sample_transform(self):
        return IceVisionTransformAdapter(
            [*A.aug_tfms(size=self.image_size), A.Normalize(), alb.RandomBrightnessContrast()]
        )


datamodule = ObjectDetectionData.from_coco(
    train_folder="data/coco128/images/train2017/",
    train_ann_file="data/coco128/annotations/instances_train2017.json",
    val_split=0.1,
    transform=BrightnessContrastTransform,
    batch_size=4,
)
```

### Environment

 - OS (e.g., Linux):
 - Python version:
 - PyTorch/Lightning/Flash Version (e.g., 1.10/1.5/0.7):
 - GPU models and configuration:
 - Any other relevant information:

**Comment by krshrimali** [comment]
Hi, @IncubatorShokuhou - Thank you for reporting this issue. Unfortunately, I'm unable to reproduce this on the master/latest release (0.7.1). It works on my system, but happy to further investigate. Can you please share the flash version you are using? Just do: `python -c "import flash; print(flash.__version__)"`.

Also, maybe try re-installing flash? `pip uninstall lightning-flash` and then: `pip install lightning-flash`.

**Comment by IncubatorShokuhou** [comment]
> Hi, @IncubatorShokuhou - Thank you for reporting this issue. Unfortunately, I'm unable to reproduce this on the master/latest release (0.7.1). It works on my system, but happy to further investigate. Can you please share the flash version you are using? Just do: `python -c "import flash; print(flash.__version__)"`.
> 
> Also, maybe try re-installing flash? `pip uninstall lightning-flash` and then: `pip install lightning-flash`.

My flash version is `0.7.1`. I tried to reinstall flash but the error still occured. I also tried to create a new conda environment and install flash, but this error still occured.

**Comment by krshrimali** [comment]
Thanks for the info and apologies for the trouble caused, @IncubatorShokuhou - I'll try reproducing and get back to you.

**Comment by ligaz** [comment]
There is a breaking change in the latest version of flash that unifies all transforms in a single API implemented in https://github.com/PyTorchLightning/lightning-flash/pull/1233. Please refer to the `0.7.1` version of the docs in order to make this work - https://lightning-flash.readthedocs.io/en/0.7.1/reference/object_detection.html

In your particular case you should use `train_transform` instead of `transform` (the new API) in `ObjectDetectionData.from_coco` call.

**Comment by IncubatorShokuhou** [comment]
> There is a breaking change in the latest version of flash that unifies all transforms in a single API implemented in #1233. Please refer to the `0.7.1` version of the docs in order to make this work - https://lightning-flash.readthedocs.io/en/0.7.1/reference/object_detection.html
> 
> In your particular case you should use `train_transform` instead of `transform` (the new API) in `ObjectDetectionData.from_coco` call.

Thank you @ligaz , it works.

**Comment by krshrimali** [comment]
> There is a breaking change in the latest version of flash that unifies all transforms in a single API implemented in #1233. Please refer to the `0.7.1` version of the docs in order to make this work - https://lightning-flash.readthedocs.io/en/0.7.1/reference/object_detection.html
> 
> In your particular case you should use `train_transform` instead of `transform` (the new API) in `ObjectDetectionData.from_coco` call.

Ah, yes, not sure why it was passing for me on the local install. Must be some env issue. Thanks @ligaz 🚀

**Comment by krshrimali** [comment]
> > There is a breaking change in the latest version of flash that unifies all transforms in a single API implemented in #1233. Please refer to the `0.7.1` version of the docs in order to make this work - https://lightning-flash.readthedocs.io/en/0.7.1/reference/object_detection.html
> > In your particular case you should use `train_transform` instead of `transform` (the new API) in `ObjectDetectionData.from_coco` call.
> 
> Thank you @ligaz , it works.

If you are happy with it, please feel free to close this issue :) Thank you for your patience. :)

**Comment by IncubatorShokuhou** [comment]
> > > There is a breaking change in the latest version of flash that unifies all transforms in a single API implemented in #1233. Please refer to the `0.7.1` version of the docs in order to make this work - https://lightning-flash.readthedocs.io/en/0.7.1/reference/object_detection.html
> > > In your particular case you should use `train_transform` instead of `transform` (the new API) in `ObjectDetectionData.from_coco` call.
> > 
> > 
> > Thank you @ligaz , it works.
> 
> If you are happy with it, please feel free to close this issue :) Thank you for your patience. :)

Thank you @krshrimali.
